waspc/data/Generator/templates/server/src/jobs/PassthroughJobFactory.js

```diff
@@ -0,0 +1,40 @@
+import { sleep } from '../utils.js'
+
+/**
+ * "Immutable-ish" passthrough job wrapper, mainly to be used for testing.
+ */
+class PassthroughJob {
+  constructor(values) {
+    this.perform = () => { }
+    this.delayMs = 0
+    Object.assign(this, values)
+  }
+
+  delay(ms) {
+    return new PassthroughJob({ ...this, delayMs: ms })
+  }
+
+  performAsync(args) {
+    const clonedJob = new PassthroughJob({ ...this })
+    return clonedJob._performAsync(args)
+  }
+
+  _performAsync(args) {
+    async function fn(delayMs, perform) {
+      if (delayMs > 0) {
+        await sleep(delayMs)
+      }
+      return perform(args)
+    }
+
+    const res = fn(this.delayMs, this.perform)
+
+    return {
+      result: async () => await res
+    }
+  }
+}
+
+export default function (fn) {
```

Comment:
I am using default here @sodic, but just read your issue about default vs named 😅 Worth changing? I can if that is the direction, and I defer to you all on JS 👍🏻

Reply:
I am also doing this in `_jobs.js` but mainly just so I can export an instance of the class.